website/source/docs/auth/aws-iam.html.md

```diff
@@ -0,0 +1,606 @@
+
+* What type of entity is authenticated:
+  * The AWS-EC2 backend authenticates AWS EC2 instances only.
+  * The AWS-IAM backend authenticates AWS IAM principals. This can include
+    IAM users, IAM roles assumed from other accounts, AWS Lambdas that are
+    launched in an IAM role, or even EC2 instances that are launched in an
+    EC2 instance profile.
+* How the entities are authenticated
+  * The AWS-EC2 backend authenticates instances by making use of the EC2 instance
+    identity document, which is a cryptographically signed document containing
+    metadata about the instance. This document changes relatively infrequently,
+    so Vault adds a number of other constructs to mitigate against replay
+    attacks, such as client nonces, role tags, instance migrations, etc.
+  * The AWS-IAM backend authenticates by having clients provide a specially
+    signed AWS API request which the backend then passes on to AWS to validate
+    the signature and tell Vault who created it. The actual secret (i.e.,
+    the AWS secret access key) is never transmitted over the wire, and the
+    AWS signature algorithm automatically expires requests after 15 minutes,
+    providing simple and robust protection against replay attacks.
+* Specific use cases
+  * If you have a long-lived EC2 instance which you are unable to relaunch into
+    an IAM instance profile, then the AWS-EC2 backend is probably the best
+    solution for you. (While you could store long-lived AWS IAM user
+    credentials on disk and use those to authenticate to Vault, that would not
+    be recommended.)
+  * If you have non-EC2 instance entities, such as IAM users, Lambdas in IAM
+    roles, or developer laptops using [AdRoll's Hologram](https://github.com/AdRoll/hologram)
+    then you would need to use the AWS-IAM backend.
+  * If you have EC2 instances which are already in an IAM instance profile, then
+    you could use either backend.
+
+## Authentication Workflow
+
+AWS has introduced the [`sts:GetCallerIdentity`](http://docs.aws.amazon.com/STS/latest/APIReference/API_GetCallerIdentity.html)
+API method to allow you to validate the identity of a caller. The client signs
+a `GetCallerIdentity` query using the [AWS Signature v4
+algorithm](http://docs.aws.amazon.com/general/latest/gr/sigv4_signing.html) and
+submits that signed query to the Vault server. The Vault server then forwards
+it on to the AWS STS service and validates the result back. Clients don't even
+need network-level access to talk to the AWS STS API endpoint; they merely need
+to sign the credentials. However, it means that the Vault server DOES need
+network-level access to send requests to the STS endpoint.
+
+Each signed AWS request includes the current timestamp to mitigate the risk of
+replay attacks. In addition, Vault allows you to require an additional header,
+`X-Vault-Server`, to be present to mitigate against different types of replay
```

Comment:
I dislike this value; it suggests some canonical name that identifies something Vault related and I'd prefer to avoid confusion as to its purpose. `X-Vault-AWS-IAM-ID` seems like one potential better value.